pkg/provider/senhasegura/dsm/dsm.go

```diff
@@ -0,0 +1,202 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dsm
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	esv1beta1 "github.com/external-secrets/external-secrets/apis/externalsecrets/v1beta1"
+	senhaseguraAuth "github.com/external-secrets/external-secrets/pkg/provider/senhasegura/auth"
+)
+
+/*
+	DSM service for SenhaseguraProvider
+*/
+type DSM struct {
+	isoSession *senhaseguraAuth.SenhaseguraIsoSession
+}
+
+/*
+	isoDappResponse is a response object from senhasegura /iso/dapp/response (DevOps Secrets Management API endpoint)
+	Contains information about API request and Secrets linked with authorization
+*/
+type isoDappResponse struct {
+	Response struct {
+		Status    int    `json:"status"`
+		Message   string `json:"message"`
+		Error     bool   `json:"error"`
+		ErrorCode int    `json:"error_code"`
+	} `json:"response"`
+	Application struct {
+		Name        string   `json:"name"`
+		Description string   `json:"description"`
+		Tags        []string `json:"tags"`
+		System      string   `json:"system"`
+		Environment string   `json:"Environment"`
+		Secrets     []struct {
+			SecretID       string              `json:"secret_id"`
+			SecretName     string              `json:"secret_name"`
+			Identity       string              `json:"identity"`
+			Version        string              `json:"version"`
+			ExpirationDate string              `json:"expiration_date"`
+			Engine         string              `json:"engine"`
+			Data           []map[string]string `json:"data"`
+		} `json:"secrets"`
+	} `json:"application"`
+}
+
+var (
+	errCannotCreateRequest = errors.New("cannot create request to senhasegura resource /iso/dapp/application")
+	errCannotDoRequest     = errors.New("cannot do request in senhasegura, SSL certificate is valid ?")
+	errInvalidResponseBody = errors.New("invalid HTTP response body received from senhasegura")
+	errInvalidHTTPCode     = errors.New("received invalid HTTP code from senhasegura")
+	errApplicationError    = errors.New("received application error from senhasegura")
+)
+
+/*
+	New creates an senhasegura DSM client based on ISO session
+*/
+func New(isoSession *senhaseguraAuth.SenhaseguraIsoSession) (*DSM, error) {
+	return &DSM{
+		isoSession: isoSession,
+	}, nil
+}
+
+/*
+	GetSecret implements ESO interface and get a single k/v pair from senhasegura provider with DSM service
+*/
+func (dsm *DSM) GetSecret(ctx context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) (resp []byte, err error) {
+	appSecrets, err := dsm.fetchSecrets()
+	if err != nil {
+		return []byte(""), err
+	}
+
+	for _, v := range appSecrets.Application.Secrets {
+		if ref.Key == v.Identity {
+			for _, v2 := range v.Data {
+				for k, v3 := range v2 {
+					if k == ref.Property {
```

Comment:
Property is not required. If no Property is provided, it should return the whole data content.

Reply:
Fixed, thanks!